core/tee/tee_fs_key_manager.c

```diff
@@ -403,5 +404,140 @@ TEE_Result tee_fs_decrypt_file(enum tee_fs_file_type file_type,
 			cipher, cipher_size, plaintext, plaintext_size);
 }
 
+static TEE_Result sha256(uint8_t *out, size_t out_size, const uint8_t *in,
+			 size_t in_size)
+{
+	TEE_Result res;
+	uint8_t *ctx = NULL;
+	size_t ctx_size;
+	uint32_t algo = TEE_ALG_SHA256;
+
+	res = crypto_ops.hash.get_ctx_size(algo, &ctx_size);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	ctx = malloc(ctx_size);
+	if (!ctx)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = crypto_ops.hash.init(ctx, algo);
+	if (res != TEE_SUCCESS)
+		goto out;
+
+	res = crypto_ops.hash.update(ctx, algo, in, in_size);
+	if (res != TEE_SUCCESS)
+		goto out;
+
+	res = crypto_ops.hash.final(ctx, algo, out, out_size);
+
+out:
+	free(ctx);
+	return res;
+}
+
+static TEE_Result aes_ecb(uint8_t *out, size_t out_size, const uint8_t *in,
+			  size_t in_size, const uint8_t *key, size_t key_size)
+{
+	TEE_Result res;
+	uint8_t *ctx = NULL;
+	size_t ctx_size;
+	uint32_t algo = TEE_ALG_AES_ECB_NOPAD;
+
+	TEE_ASSERT(out_size == in_size);
+
+	res = crypto_ops.cipher.get_ctx_size(algo, &ctx_size);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	ctx = malloc(ctx_size);
+	if (!ctx)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	res = crypto_ops.cipher.init(ctx, algo, TEE_MODE_ENCRYPT, key,
+				     key_size, NULL, 0, NULL, 0);
+	if (res != TEE_SUCCESS)
+		goto out;
+
+	res = crypto_ops.cipher.update(ctx, algo, TEE_MODE_ENCRYPT, true, in,
+				       in_size, out);
+	if (res != TEE_SUCCESS)
+		goto out;
+
+	crypto_ops.cipher.final(ctx, algo);
+	res = TEE_SUCCESS;
+
+out:
+	free(ctx);
+	return res;
+}
+
+static TEE_Result essiv(uint8_t *iv, size_t iv_size, const uint8_t *fek,
```

Comment:
Since this code isn't flexible about the sizes of iv and fek supply them as arrays of `TEE_AES_BLOCK_SIZE` and `TEE_FS_KM_FEK_SIZE` and save some parameters (and TEE_ASSERT).